zerver/lib/remote_server.py

```diff
@@ -226,6 +229,16 @@ def send_to_push_bouncer(
             # The callers requesting this endpoint want the exception to propagate
             # so they can catch it.
             raise RemoteRealmServerMismatchError
+        elif endpoint == "push/e2ee/register" and "code" in result_dict:  # nocoverage
+            code = result_dict["code"]
+            assert code in ["INVALID_BOUNCER_PUBLIC_KEY", "BAD_REQUEST", "REQUEST_EXPIRED"]
+
+            if code == "INVALID_BOUNCER_PUBLIC_KEY":
+                raise InvalidBouncerPublicKeyError
+            elif code == "BAD_REQUEST":
+                raise InvalidEncryptedPushRegistrationError
+            elif code == "REQUEST_EXPIRED":
+                raise RequestExpiredError
         else:
             # But most other errors coming from the push bouncer
             # server are client errors (e.g. never-registered token)
```

Comment:
Hmm, does `BAD_REQUEST` actually need separate handling separate from the `JsonableError` fallthrough below?

Reply:
It was needed earlier. 

But with the cleanup we did in 6th commit removes the need.